site/bulletin_board/routes/users.js

```diff
@@ -84,4 +86,26 @@ router.post('/', (req, res, next) => {
   });
 });
 
+router.get('/profile', (req, res, next) => {
+  res.redirect("profile/"+req.session.passport.user);
+});
+router.get('/profile/:id', (req, res, next) => {
+  datasource.retrieve(req.params['id'], req.user.id, (user) => {
+    postsourece.retrieveperuser(req.user.id, (posts) => {
+      postsourece.post_liked(req.user.id, (posts_liked) => {
+        console.log(posts)
+
+        res.render('profile_view', { name : user.username, birthdate : user.birthdate , user: user ,posts : posts,posts_liked:posts_liked});
+      });});
+  });
+});
+
+router.get('/:id', (req, res, next) => {
```

Comment:
It's totally reasonable to have two different routes for viewing vs. editing.. but worth considering the alternative of a single endpoint.